### PR TITLE
ft: check all mandatory option items logic

### LINKED ISSUE #2: Mandatory choice filled but error message is still shown
## Reproduce step:
- Input URL: https://shopeefood.vn/ho-chi-minh/com-tam-phuc-loc-tho-bau-cat
- Get menu
- Create invitation
- Add `Combo Sườn Tiết Kiệm + 2 Món Tùy Chọn` to cart => error shown

## Expected behavior:
- When choosing 0 option, error is shown (must choose 2 according to min and max)
- When choosing 1 option, error is shown (must choose 2 according to min and max)
- When choosing 2 options, can be proceed

## Checklist - copy to PR
- [ ] Expected behavior is met
- [ ] Validate min and max quantity based on `weight` of option (good to have)
- [ ] Replicate both logics to `audit` in api `handlers/order` (good to have)

## Instructions:
- For original work, you can start from this: [https://github.com/lamnguyencse17/foodgether/blob/main/src/components/invitation/ItemOptionModal.tsx](https://github.com/lamnguyencse17/foodgether/blob/main/src/components/invitation/ItemOptionModal.tsx)

### PR DESCRIPTION
**Pull request by nngthinh**
### Description
Issue #2 

- Check logic for all mandatory option items:
  - Mandatory item: exist in `menuOption`
  - List of choice items: exist in `menuOption` & have quantity in the allowed range

### CODE REVIEW

File: src/components/invitation/ItemOptionModal.tsx [hunk 1]
```diff
@@ -62,6 +62,12 @@ const ItemOptionModal: FunctionComponent<ItemOptionModalProps> = ({
     shallow
   );
 
+  const currentDishOptionMap = useMemo(() => currentDishOption.reduce((obj, detail) => {
+    obj[detail.optionId] = detail;
+    return obj;
+  }, {} as Record<string, DishOptionValue>)
+  , [currentDishOption]);
```
Comment (lamnguyencse17) [inline src/components/invitation/ItemOptionModal.tsx]: Please try using [objectify from radash instead](https://radash-docs.vercel.app/docs/array/objectify)

File: src/components/invitation/ItemOptionModal.tsx [hunk 2]
```diff
@@ -77,13 +83,27 @@ const ItemOptionModal: FunctionComponent<ItemOptionModalProps> = ({
...
+          selectedOptionItems.length >= minQuantity 
+          && selectedOptionItems.length <= maxQuantity
+          && selectedOptionItems.every(({ optionItemId }) => !!get(menuOption.items, optionItemId.toString()))
+        ) {
+          return;
+        }
+      } else if (
+        !!selectedOptionItems 
+        && get(menuOption.items, selectedOptionItems.optionItemId?.toString())
+      ) { // mandatory
```
Comment (lamnguyencse17) [inline src/components/invitation/ItemOptionModal.tsx]: you can try to unify `selectedOptionItems` to array to eliminate the need to check for this

File: src/components/invitation/ItemOptionModal.tsx [hunk 3]
```diff
@@ -77,13 +83,27 @@ const ItemOptionModal: FunctionComponent<ItemOptionModalProps> = ({
       {}
     ) as OptionDictOptionData;
     let haveAllMandatoryOption = true;
-    listify(dishOption, (_, value) => ({ ...value })).forEach((menuOption) => {
+    listify(dishOption, (key, value) => ({ optionId: key, ...value })).forEach((menuOption) => {
```
Comment (lamnguyencse17) [inline src/components/invitation/ItemOptionModal.tsx]: Technically optionId was already in `value` please check dishOption again to verify

**Review by lamnguyencse17 — CHANGES_REQUESTED** [review]

**Comment by lamnguyencse17** [comment]
Commit cherry picked from `main` branch to fix linting on PR

File: src/components/invitation/ItemOptionModal.tsx (hunk 1 above)
Comment (nngthinh) [reply src/components/invitation/ItemOptionModal.tsx]: Fixed

File: src/components/invitation/ItemOptionModal.tsx (hunk 2 above)
Comment (nngthinh) [reply src/components/invitation/ItemOptionModal.tsx]: Fixed

File: src/components/invitation/ItemOptionModal.tsx (hunk 3 above)
Comment (nngthinh) [reply src/components/invitation/ItemOptionModal.tsx]: Fixed

**Pull request merged**